src/modules/tests/test-data/cat-adi-part2/manoeuvres/manoeuvres.reducer.ts

```diff
@@ -1,14 +1,88 @@
 import { CatADI2UniqueTypes } from '@dvsa/mes-test-schema/categories/ADI2';
-import * as manoeuvresCatADI2Actions from './manoeuvres.actions';
+import * as manoeuvresActions from './manoeuvres.actions';
+import { CompetencyOutcome } from '../../../../../shared/models/competency-outcome';
 
-export const initialState: CatADI2UniqueTypes.Manoeuvres[] = [];
+export const initialState: CatADI2UniqueTypes.Manoeuvres[] = [{}, {}];
+
+let newState = initialState;
 
-// TODO - ADI2: impliment manoeuvres reducer
 export function manoeuvresCatADI2Reducer(
   state = initialState,
-  action: manoeuvresCatADI2Actions.Types,
+  action: manoeuvresActions.Types,
 ): CatADI2UniqueTypes.Manoeuvres[] {
   switch (action.type) {
+    case manoeuvresActions.RECORD_MANOEUVRES_SELECTION:
+      newState = state.slice(0);
+
+      newState[action.index] = {
+        [action.manoeuvre]: {
+          ...state[action.index][action.manoeuvre],
+          selected: true,
+        },
+      };
+
+      if (
```

Comment:
Can we not have if statements in a reducer please?

Reply:
Why not if the reducer remains pure and immutable?

Reply:
Because we use reducers to update state and only to update state. Build new actions if needed and move the `if` statement where the actions are dispatched

Reply:
Absolutely agree, and it does. 
In my experience Actions represent what happened not what should happen which is what, I think, you're suggesting. If this precedent already exists then I will, of course, follow existing practice.

Reply:
Me personally, I would not like to work with reducer in the future. As it's difficult to reason about and to understand what's happening. 

Reducers should be the most easy to understand. And I am pretty sure there are ways to implement this with less complexity.

If you write unit tests which gives full coverage of this reducer I'm happy to resolve this conversation

Reply:
Moved reducer logic in to page component.